### PR TITLE
Tickets/dm33151: Added Track Target, Prepare for on Sky and Shutdown/Enable tests

### PR DESCRIPTION
**Pull request by rbovill**

### CODE REVIEW

File: bin/auxtel_prepare_for_onsky.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+
+from lsst.ts.IntegrationTests import AuxTelPrepare
```
Comment (mareuter) [inline bin/auxtel_prepare_for_onsky.py:25]: We have a SQ script called prepare_for_flat that we'll want to run, so we need this class disambiguated. Call it `AuxTelPrepareOnSky`

File: python/lsst/ts/IntegrationTests/aux_tel_prepare_for_onsky.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["AuxTelPrepare"]
+
+from lsst.ts.IntegrationTests import BaseScript
+
+
+class AuxTelPrepare(BaseScript):
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/aux_tel_prepare_for_onsky.py:26]: See comment above about class name.

File: tests/test_auxtel_prepare_for_onsky.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+
+from lsst.ts import salobj
+from lsst.ts.IntegrationTests import ScriptQueueController
+from lsst.ts.IntegrationTests import AuxTelVisit
+
+
+class AuxTelVisitTestCase(unittest.IsolatedAsyncioTestCase):
```
Comment (mareuter) [inline tests/test_auxtel_prepare_for_onsky.py:31]: You aren't running the script class you think you are here. I think this all needs to be changed to run the AuxTelPrepareOnSky script class.

File: python/lsst/ts/IntegrationTests/aux_tel_track_target.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# This file is part of ts_IntegrationTests
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/aux_tel_track_target.py:1]: This is not what I had in mind. Each track_target script needs to be followed by the visit image sequence scripts. I think we need to talk about this one more if you don't understand what I'm after here.

**Review by mareuter — CHANGES_REQUESTED** [review]
Please check the comments as there's work to do.

File: bin/auxtel_prepare_for_onsky.py (hunk 1 above)
Comment (rbovill) [reply bin/auxtel_prepare_for_onsky.py]: Fixed

File: python/lsst/ts/IntegrationTests/aux_tel_prepare_for_onsky.py (hunk 2 above)
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/aux_tel_prepare_for_onsky.py]: Fixed

File: tests/test_auxtel_prepare_for_onsky.py (hunk 3 above)
Comment (rbovill) [reply tests/test_auxtel_prepare_for_onsky.py]: Fixed

File: python/lsst/ts/IntegrationTests/aux_tel_track_target.py (hunk 4 above)
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/aux_tel_track_target.py:1]: Fixed.  I think.

File: python/lsst/ts/IntegrationTests/aux_tel_track_target.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    parser.add_argument(
+        "-t",
+        "--target",
+        type=str,
+        required=True,
+        help="""Specify the target to track.""",
+    )
+
+    def __init__(self, isStandard=True, queue_placement="after", target=None):
+        self.parsed = type(self).parser.parse_args()
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/aux_tel_track_target.py:51]: Remove the type call as it's confusing.

File: python/lsst/ts/IntegrationTests/aux_tel_prepare_for_onsky.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
+# This file is part of ts_IntegrationTests
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/aux_tel_prepare_for_onsky.py:1]: Please normalize the file names here. The top-level scripts use auxtel but these scripts are named aux_tel. I prefer the former.

**Review by mareuter — CHANGES_REQUESTED** [review]
Getting closer. Few more things to take care of.

File: tests/test_auxtel_shutdown.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        # Set the LSST_DDS_PARTITION_PREFIX ENV_VAR.
+        salobj.set_random_lsst_dds_partition_prefix()
+
+        # Create the ScriptQueue Controller.
+        self.controller = ScriptQueueController(index=2)
+
+        # Start the controller and wait for it be ready.
+        await self.controller.start_task
+
+    async def test_auxtel_track_target(self):
```
[the remark this answers is not in the document]
Comment (rbovill) [reply tests/test_auxtel_shutdown.py:44]: Fixed

File: tests/test_auxtel_shutdown.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+import unittest
+
+from lsst.ts import salobj
+from lsst.ts.IntegrationTests import ScriptQueueController
+from lsst.ts.IntegrationTests import AuxTelShutdown
+
+
+class AuxTelShutdownTestCase(unittest.IsolatedAsyncioTestCase):
+    """Test the AuxTel Visit integration test script."""
```
[the remark this answers is not in the document]
Comment (rbovill) [reply tests/test_auxtel_shutdown.py:32]: Fixed

File: tests/test_auxtel_track_target.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+import unittest
+import sys
+
+from lsst.ts import salobj
+from lsst.ts.IntegrationTests import ScriptQueueController
+from lsst.ts.IntegrationTests import AuxTelTrackTarget
+
+
+class AuxTelTrackTargetTestCase(unittest.IsolatedAsyncioTestCase):
+    """Test the AuxTel Visit integration test script."""
```
[the remark this answers is not in the document]
Comment (rbovill) [reply tests/test_auxtel_track_target.py:33]: Fixed

File: python/lsst/ts/IntegrationTests/aux_tel_track_target.py (hunk 5 above)
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/aux_tel_track_target.py:51]: Fixed

File: python/lsst/ts/IntegrationTests/aux_tel_prepare_for_onsky.py (hunk 6 above)
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/aux_tel_prepare_for_onsky.py:1]: Fixed

**Review by mareuter — APPROVED** [review]
Looks good now. Thanks for hanging in there.

**Pull request merged**